### PR TITLE
Automatically tag PRs which may affect the C# extension.

### PR DESCRIPTION
**Pull request by JoeRobich**
It can be tedious to determine what changes in Roslyn are relevant to the C# extension. The idea is that changes under these folders may change the behavior of the Roslyn LSP and thus impact the C# extension. PR authors may remove the label if they do not believe their change is impactful.

### CODE REVIEW

File: .github/policies/resourceManagement.yml [hunk 1]
```diff
@@ -144,6 +144,28 @@ configuration:
           label: Community
       description: Label Community Pull Requests
       triggerOnOwnActions: false
+    - if:
```
Comment (333fred) [inline .github/policies/resourceManagement.yml:147]: Does this actually work? I've not see the equivalent API Owners tagging actually work well.

**Review by dibarbet — APPROVED** [review]
I'm fine with this. But definitely should get thoughts from other @dotnet/roslyn-infrastructure folks

File: .github/policies/resourceManagement.yml (hunk 1 above)
Comment (JoeRobich) [reply .github/policies/resourceManagement.yml:147]: I have no idea. I based it on the other PR taggers, but am not sure there is a good way to test.

Comment (JoeRobich) [reply .github/policies/resourceManagement.yml:147]: Looking at [docs](https://eng.ms/docs/more/github-inside-microsoft/policies/examples/pr-triage) and I think the indentation is all wrong. Will update.

File: .github/policies/resourceManagement.yml [hunk 2]
```diff
@@ -8,6 +8,7 @@ where:
 configuration:
   resourceManagementConfiguration:
     scheduledSearches:
+
```
Comment (JoeRobich) [inline .github/policies/resourceManagement.yml:11]: Added empty line between searches and tasks

File: .github/policies/resourceManagement.yml [hunk 3]
```diff
@@ -23,8 +24,12 @@ configuration:
       - closeIssue
       - addReply:
           reply: "Closing this issue as we've seen no reply to the request for more information. If you are able to get the requested information, please add it to the issue and we will retriage it. "
+
     eventResponderTasks:
-    - if:
+
+    - description: Auto-approve auto-merge PRs
+      triggerOnOwnActions: false
```
Comment (JoeRobich) [inline .github/policies/resourceManagement.yml:31]: Moved description and trigger to the top of each task to improve readability

File: .github/policies/resourceManagement.yml [hunk 4]
```diff
@@ -187,12 +222,14 @@ configuration:
...
       - addLabel:
           label: Needs UX Triage
-      description: Add "Needs UX Triage" on PRs
+
+    - description: Adds "Needs API Review" on PRs that touch public APIs
       triggerOnOwnActions: false
-    - if:
+      if:
       - payloadType: Pull_Request
+      - isPullRequest
```
Comment (JoeRobich) [inline .github/policies/resourceManagement.yml:230]: Added isPullRequest filter where applicable

File: .github/policies/resourceManagement.yml (hunk 1 above)
Comment (JoeRobich) [reply .github/policies/resourceManagement.yml]: Doesn't look like the formatting was _that_ different after all. Still made changes to improve readability.

**Comment by JoeRobich** [comment]
Gonna get this merged so that I can test it. If the tagging doesn't work then it is an easy revert.

**Pull request merged**